internal/testrunner/script/script.go

```diff
@@ -0,0 +1,688 @@
+
+package script
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/rogpeppe/go-internal/testscript"
+	"github.com/spf13/cobra"
+
+	"github.com/elastic/elastic-package/internal/cobraext"
+	"github.com/elastic/elastic-package/internal/configuration/locations"
+	"github.com/elastic/elastic-package/internal/elasticsearch/ingest"
+	"github.com/elastic/elastic-package/internal/packages"
+	"github.com/elastic/elastic-package/internal/packages/changelog"
+	"github.com/elastic/elastic-package/internal/resources"
+	"github.com/elastic/elastic-package/internal/servicedeployer"
+	"github.com/elastic/elastic-package/internal/stack"
+)
+
+func Run(dst io.Writer, cmd *cobra.Command, args []string) error {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("could not find home: %w", err)
+	}
+	loc, err := locations.NewLocationManager()
+	if err != nil {
+		return err
+	}
+	work, err := cmd.Flags().GetBool(cobraext.WorkScriptTestFlagName)
+	if err != nil {
+		return err
+	}
+	workRoot := filepath.Join(home, filepath.FromSlash(".elastic-package/tmp/script_tests"))
+	err = os.MkdirAll(workRoot, 0o700)
+	if err != nil {
+		return fmt.Errorf("could not make work space root: %w", err)
+	}
+	var workdirRoot string
+	if work {
+		// Only create a work root and pass it in if --work has been requested.
+		// The behaviour of testscript is to set TestWork to true if the work
+		// root is non-zero, so just let testscript put it where it wants in the
+		// case that we have not requested work to be retained. This will be in
+		// os.MkdirTemp(os.Getenv("GOTMPDIR"), "go-test-script") which on most
+		// systems will be /tmp/go-test-script. However, due to… decisions, we
+		// cannot operate in that directory…
+		workdirRoot, err = os.MkdirTemp(workRoot, "*")
+		if err != nil {
+			return fmt.Errorf("could not make work space: %w", err)
+		}
+	} else {
+		// … so set $GOTMPDIR to a location that we can work in.
+		//
+		// This is all obviously awful.
+		err = os.Setenv("GOTMPDIR", workRoot)
+		if err != nil {
+			return fmt.Errorf("could not set temp dir var: %w", err)
+		}
+	}
+
+	externalStack, err := cmd.Flags().GetBool(cobraext.ExternalStackFlagName)
+	if err != nil {
+		return err
+	}
+	run, err := cmd.Flags().GetString(cobraext.RunPatternFlagName)
+	if err != nil {
+		return err
+	}
+	verbose, err := cmd.Flags().GetCount(cobraext.VerboseFlagName)
+	if err != nil {
+		return err
+	}
+	verboseScript, err := cmd.Flags().GetBool(cobraext.VerboseScriptFlagName)
+	if err != nil {
+		return err
+	}
+	update, err := cmd.Flags().GetBool(cobraext.UpdateScriptTestArchiveFlagName)
+	if err != nil {
+		return err
+	}
+	cont, err := cmd.Flags().GetBool(cobraext.ContinueOnErrorFlagName)
+	if err != nil {
+		return err
+	}
```

Comment:
Flags should be get from `cmd` as done in other sub-commands.

Reply:
This will be done in the follow up, #3053.